### PR TITLE
990-fe-refactor-icon-props-with-favorites-icon

### PR DESCRIPTION
**Pull request by REALcuba**

### CODE REVIEW

File: packages/ui/src/components/atoms/Icon.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import React from 'react'
 import styled from 'styled-components'
 
-interface IconStyledProps extends TIcon {
-  $wght: number
-  $grad: number
-  $opsz: number
-}
+// interface IconStyledProps extends TIcon {
```
Comment (kevinmamaqi) [inline packages/ui/src/components/atoms/Icon.tsx:4]: Eliminar

File: packages/ui/src/components/atoms/Icon.tsx [hunk 2]
```diff
@@ -28,16 +28,20 @@ export type TIcon = React.HTMLAttributes<HTMLSpanElement> & {
   className?: string
   color?: string
   cursor?: string
+  isFavorite?: boolean,
```
Comment (kevinmamaqi) [inline packages/ui/src/components/atoms/Icon.tsx:31]: Eliminar

File: packages/ui/src/components/atoms/Icon.tsx [hunk 3]
```diff
@@ -28,16 +28,20 @@ export type TIcon = React.HTMLAttributes<HTMLSpanElement> & {
...
+  onClick?: () => void
 }
 
 export const Icon: React.FC<TIcon> = ({
   className = '',
-  fill = 1,
```
Comment (kevinmamaqi) [inline packages/ui/src/components/atoms/Icon.tsx:35]: Pasar con valor por defecto, igual que wght, grad, opsz...

File: packages/ui/src/stories/atoms/Icon.stories.ts [hunk 4]
```diff
@@ -13,32 +14,39 @@ const meta = {
     onClick: {
       action: 'clicked',
     },
+    isFavorite: {
```
Comment (kevinmamaqi) [inline packages/ui/src/stories/atoms/Icon.stories.ts:17]: Eliminar

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: packages/ui/src/__tests__/atoms/Icon.test.tsx [hunk 5]
```diff
@@ -17,8 +17,23 @@ describe('Icon', () => {
...
     expect(icon).toHaveStyle(
-      `font-variation-settings: 'FILL' 1, 'wght' 400,'GRAD' 0, 'opsz' 48;`
+      `font-variation-settings: 'FILL' 0, 'wght' 400,'GRAD' 0, 'opsz' 48;`
     )
     expect(icon).toHaveStyle(`color: ${colors.primary}`)
   })
+  it('renders correctly with fill = 1', () => {
+    render(
+      <Icon data-testid="icon" name="favorite" onClick={() => { }} $fill={1}
+      >
```
Comment (marmorrei) [inline packages/ui/src/__tests__/atoms/Icon.test.tsx:27]: Is it necessary to have the $fill prop here? Is it not changing when you do the fireEvent.click(icon) to 1?

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: packages/ui/src/__tests__/atoms/Icon.test.tsx [hunk 6]
```diff
@@ -17,8 +17,23 @@ describe('Icon', () => {
...
+        favorite
+      </Icon>
+    )
+    const icon = screen.getByTestId('icon')
+    expect(icon).toBeInTheDocument();
+    expect(icon).toHaveStyle(
+      `font-variation-settings: 'FILL' 1, 'wght' 400,'GRAD' 0, 'opsz' 48;`
+    )
+    expect(icon).toHaveAttribute('name', 'favorite');
+    fireEvent.click(icon);
```
Comment (NataliaR-BCN) [inline packages/ui/src/__tests__/atoms/Icon.test.tsx:37]: Mock the onClick function and test if it is been called with this click

File: packages/ui/src/__tests__/atoms/Icon.test.tsx (hunk 5 above)
Comment (REALcuba) [reply packages/ui/src/__tests__/atoms/Icon.test.tsx:27]: yes, because the value of  the prop $fill need to be passed, so it will render filled or not

**Review by NataliaR-BCN — APPROVED** [review]

**Review by REALcuba — COMMENTED** [review]
done

**Review by kevinmamaqi — APPROVED** [review]

**Pull request merged**